### ISSUE TITLE
java.lang.NoClassDefFoundError: java/sql/ShardingKey

### ISSUE THREAD

**Issue by xuwei-k** [brettwooldridge/HikariCP#1578]
`java.sql.ShardingKey` does not exists in JDK8.

I think something wrong in build settings in HikariCP 3.4.4 🤔 

https://github.com/brettwooldridge/HikariCP/compare/HikariCP-3.4.3...HikariCP-3.4.4

version: 3.4.4
JDK: 8

- https://travis-ci.com/github/playframework/playframework/jobs/329797011
- https://github.com/playframework/playframework/pull/10264

```
[error] Test play.db.jpa.JPAApiTest.shouldWorkWithEmptyConfiguration failed: java.lang.NoClassDefFoundError: java/sql/ShardingKey, took 1.61 sec
[error]     at java.lang.Class.getDeclaredMethods0(Native Method)
[error]     at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
[error]     at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
[error]     at java.lang.Class.getMethod0(Class.java:3018)
[error]     at java.lang.Class.getMethod(Class.java:1784)
[error]     at org.hibernate.engine.jdbc.env.internal.DefaultSchemaNameResolver.determineAppropriateResolverDelegate(DefaultSchemaNameResolver.java:44)
[error]     at org.hibernate.engine.jdbc.env.internal.DefaultSchemaNameResolver.resolveSchemaName(DefaultSchemaNameResolver.java:75)
[error]     at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentImpl.determineCurrentSchemaName(JdbcEnvironmentImpl.java:311)
[error]     at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentImpl.<init>(JdbcEnvironmentImpl.java:266)
[error]     at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:114)
```

- https://github.com/xuwei-k/scalikejdbc/commit/b2d1e7d1ec1e42a393ad50e89a51160a8857862c
- https://travis-ci.com/github/xuwei-k/scalikejdbc/jobs/329842154

```
[error] Uncaught exception when running scalikejdbc.ConnectionPoolsSpec: java.lang.NoClassDefFoundError: java/sql/ShardingKey
[error] sbt.ForkMain$ForkError: java.lang.NoClassDefFoundError: java/sql/ShardingKey
[error] 	at java.lang.Class.getDeclaredMethods0(Native Method)
[error] 	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
[error] 	at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
[error] 	at java.lang.Class.getMethod0(Class.java:3018)
[error] 	at java.lang.Class.getMethod(Class.java:1784)
[error] 	at scalikejdbc.LoanPattern.reflMethod$Method1(LoanPattern.scala:25)
[error] 	at scalikejdbc.LoanPattern.using(LoanPattern.scala:25)
[error] 	at scalikejdbc.LoanPattern.using$(LoanPattern.scala:20)
[error] 	at scalikejdbc.NamedDB.using(NamedDB.scala:20)
[error] 	at scalikejdbc.DBConnection.readOnly(DBConnection.scala:210)
[error] 	at scalikejdbc.DBConnection.readOnly$(DBConnection.scala:209)
[error] 	at scalikejdbc.NamedDB.readOnly(NamedDB.scala:20)
[error] 	at scalikejdbc.ConnectionPoolsSpec.$anonfun$new$1(ConnectionPoolsSpec.scala:58)
[error] 	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
```

**Comment by rnorth** [comment]
Just to add another data point, we upgraded from 3.4.3 to 3.4.4 as well. We're seeing the same exception on our JDK8 builds, but building on JDK11 is absolutely fine.

**Comment by brettwooldridge** [comment]
@xuwei-k @rnorth Understood.  The issue is my fault.  Compiling the library under JDK 11, even with JDK 8 as the target, is causing the issue.  The generated proxy classes are built using reflection and therefore wind up with JDK 11 class references.

I am investigating how to fix the maven build now and will publish a new version as soon as the issue is resolved.

**Comment by mkurz** [comment]
@brettwooldridge Maybe `javac --release 8` helps? See https://stackoverflow.com/a/43103038/810109

**Comment by brettwooldridge** [comment]
Fixed and v3.3.5 published.
